include/LightGBM/c_api.h

```diff
@@ -724,7 +724,7 @@ LIGHTGBM_C_EXPORT int LGBM_FastConfigFree(FastConfigHandle fastConfig);
  * \note
  * You should pre-allocate memory for ``out_result``:
  *   - for normal and raw score, its length is equal to ``num_class * num_data``;
- *   - for leaf index, its length is equal to ``num_class * num_data * num_iteration``;
+ *   - for leaf index, its length is equal to ``num_class * num_data * (start_iteration + num_iteration)``;
```

Comment:
I think here should be  `* num_iteration` instead of `* (start_iteration + num_iteration)`. Since in `PredictLeafIndex` in gbdt_prediction.cpp, we only output the leaf indices from `start_iteration`.

Reply:
is `num_iteration` the count, or the `end_iteration`?

Reply:
It's the count.

Reply:
@shiyu1994 Thanks a lot! Indeed, those `start_iteration +` changes should be reverted. Initially, when I started to read your original PR I thought that `num_iteration` means the last used iteration (just like in `dump_model` or `save_model`). But after getting to cpp files I understood that it is number of iteration actually and forgot to revert docs changes I had already done.

Reply:
Done in f55619d3b8a2bc8a0dff4eccaa9c957e07e1272d.